plugins/available/blesh.plugin.bash

```diff
@@ -0,0 +1,19 @@
+# shellcheck shell=bash
+cite about-plugin
+about-plugin 'load ble.sh, the Bash line editor!'
+
+if [[ ${BLE_VERSION-} ]]; then
+	_log_warning "ble.sh is already loaded!"
+	return
+fi
+
+_bash_it_ble_path=${XDG_DATA_HOME:-$HOME/.local/share}/blesh/ble.sh
```

Comment:
Should we check for `ble.sh` already being in $PATH?

Reply:
Maybe, though I didn't expect that `~/.local/share/blesh`, etc. is added to `PATH`.

One problem is that `ble.sh` found in `PATH` can be an unrelated script with the same filename as Bash Line Editor. There is no other public tool of the name `ble.sh` as far as I know, but users of Bluetooth Low Energy (BLE) seem to name their tiny scriptlet to start Bluetooth `ble.sh` or `start_ble.sh`, etc., which I haven't noticed when I determined the name `ble.sh`.

Reply:
@akinomyoga interesting problem. I only brought it up because usually tools that require `make` eventually end up as distribution packages, and thus end up somewhere in $PATH. We can leave this issue aside for now, until someone brings it up.

Reply:
I don't thing it makes too much sense to search in `$PATH`, as we do not do that in other plugins with similar motives

Reply:
Yeah, `ble.sh` uses `make` to combine several script files to form a single core script `ble.sh`, but it is still a script for `source` (The reason I use `make` is just that I don't want to directly edit a single solid script file of 30k lines). Thus, it will be installed on `$prefix/share/blesh`. For example, the AUR package of `ble.sh` for Arch Linux installs `ble.sh` in `/usr/share/blesh` but not in `/usr/bin`.

I think a comparable case is `bash-completion`. The package of `bash-completion` is found in every distribution, but the script is usually installed in `/usr/share/bash-completion` with an entry point put in `/etc/profile.d`. The paths `/usr/share/bash-completion` and `/usr/profile.d` are not included in `PATH`, and I think nothing is installed in `/usr/bin` by `bash-completion`.

Reply:
Hmm, in that sense, I think we can also test `blesh/ble.sh` in `${XDG_DATA_DIRS:-/usr/local/share:/usr/share}` [ Note: this needs to be split by colons `:` and each path should be checked ] when `${XDG_DATA_HOME:-$HOME/.local/share}/blesh/ble.sh` is not found.

Reply:
@NoahGorny FWIW, I think we do this in almost every plugin by first checking with `_command_exists` before `pathmunge`.

Reply:
> Yeah, `ble.sh` uses `make` to combine several script files to form a single core script `ble.sh`, but it is still a script for `source` (The reason I use `make` is just that I don't want to directly edit a single solid script file of 30k lines). Thus, it will be installed on `$prefix/share/blesh`. For example, the AUR package of `ble.sh` for Arch Linux installs `ble.sh` in `/usr/share/blesh` but not in `/usr/bin`.
> 
> I think a comparable case is `bash-completion`. The package of `bash-completion` is found in every distribution, but the script is usually installed in `/usr/share/bash-completion` with an entry point put in `/etc/profile.d`. The paths `/usr/share/bash-completion` and `/usr/profile.d` are not included in `PATH`, and I think nothing is installed in `/usr/bin` by `bash-completion`.

works for me